hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/scheduler/capacity/AbstractLeafQueue.java

```diff
@@ -1693,12 +1693,16 @@ protected boolean canAssignToUser(Resource clusterResource,
 
   @Override
   protected void setDynamicQueueProperties() {
-    // set to -1, to disable it
-    queueContext.getConfiguration().setUserLimitFactor(getQueuePath(), -1);
-    // Set Max AM percentage to a higher value
-    queueContext.getConfiguration().setMaximumApplicationMasterResourcePerQueuePercent(
-        getQueuePath(), 1f);
     super.setDynamicQueueProperties();
+    // set to -1, to disable it, if not already set
+    CapacitySchedulerConfiguration conf = queueContext.getConfiguration();
```

Comment:
I'm not sure about this one, why we do we need to set these defaults only when it was not set before?

Why the original code no longer works?

Reply:
The original code only worked because the config changes were not reflected in ConfigurationProperties. Since this PR is intended to remedy that issue, we need to address this case as well.